Orange/widgets/data/owdatasets.py

```diff
@@ -456,20 +457,24 @@ def set_model(self, model, current_index):
                 QItemSelectionModel.ClearAndSelect | QItemSelectionModel.Rows)
             self.commit()
 
+    def indicator_state_for_info(self, info):
+        localinfo = list_local(self.local_cache_path)
+        if not info.file_path in localinfo:
+            return None
+        return (
+            os.path.join(self.local_cache_path, *info.file_path)
+            == self.current_output)
+
     def __update_cached_state(self):
         model = self.view.model().sourceModel()
-        localinfo = list_local(self.local_cache_path)
         assert isinstance(model, QStandardItemModel)
         allinfo = []
         for i in range(model.rowCount()):
             item = model.item(i, 0)
             info = item.data(Qt.UserRole)
-            is_local = info.file_path in localinfo
-            is_current = (is_local and
-                          os.path.join(self.local_cache_path, *info.file_path)
-                          == self.current_output)
-            item.setData(" " * (is_local + is_current), Qt.DisplayRole)
-            item.setData(self.IndicatorBrushes[is_current], Qt.ForegroundRole)
+            state = self.indicator_state_for_info(info)
+            item.setData(state, UniformHeightIndicatorDelegate.IndicatorRole)
+            item.setData(self.IndicatorBrushes[bool(state)], Qt.ForegroundRole)
```

Comment:
The previous spaces were used for sorting, which now does not work anymore.

Reply:
Fixed, and also added a comment.